TweetBane/app/src/main/java/com/elysium/tweetbane/MainActivity.java

```diff
@@ -0,0 +1,142 @@
+        mEditText = (EditText) findViewById(R.id.input);
+        mButton = (Button) findViewById(R.id.button);
+        mRecyclerView.setLayoutManager(new LinearLayoutManager(this,LinearLayoutManager.VERTICAL,false));
+        getAccessToken();
+
+        View.OnClickListener listener = new View.OnClickListener() {
+
+            @Override
+            public void onClick(View view) {
+                getTweet(mEditText.getText().toString());
+            }
+        };
+
+        mButton.setOnClickListener(listener);
+    }
+    private void getAccessToken(){
+        OkHttpClient client = new OkHttpClient();
+
+        RequestBody body = new FormBody.Builder()
+                .add("grant_type", "client_credentials")
+                .build();
+
+        Headers headers = new Headers.Builder()
+                .add("Authorization","Basic "+AccessToken)
+                .add("Content-Type","application/x-www-form-urlencoded;charset=UTF-8")
+                .build();
+
+        Request request = new Request.Builder().url("https://api.twitter.com/oauth2/token")
+                .headers(headers)
+                .post(body).build();
+
+        client.newCall(request).enqueue(new Callback() {
+
+            @Override
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+
+                if (!response.isSuccessful()) {
+                    throw new IOException("Unexpected Error: "+response);
+                }
+
+                try {
+
+                    JSONObject result =  new JSONObject(response.body().string());
+                    mToken = result.getString("access_token");
+
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+    }
+
+    private void getTweet(String s) {
+
+        OkHttpClient client = new OkHttpClient();
+        Headers headers = new Headers.Builder()
+                .add("Authorization","Bearer "+mToken)
+                .build();
+
+        Request request = new Request.Builder().url("https://api.twitter.com/1.1/statuses/user_timeline.json?screen_name="+s).headers(headers).build();
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                if (!response.isSuccessful()){
+                    throw new IOException("Unexpected Error: "+response);
+                }
+
+                Gson gson = new Gson();
+
+                Type listtype = new TypeToken<List<Tweets>>(){}.getType();
+                final List<Tweets> tweets = gson.fromJson(response.body().string(),listtype);
```

Comment:
I think this might be your problem - you should loop through the list of JSONObjects and call `gson.fromJson(object, Tweet.class)` on each one